lib/frontend.js

```diff
@@ -12,4 +16,55 @@ app.get('/', (req, res) => {
   res.render('index.hbs', { year: new Date().getFullYear() });
 });
 
+app.get('/github/oauth/login', async (req, res) => {
+  if (!req.query.state) {
+    return res.status(400).send('Error: State parameter was not provided');
+  }
+
+  let state;
+  try {
+    state = await SignedParams.load(req.query.state);
+  } catch (err) {
+    if (err.name && err.name === 'JsonWebTokenError') {
+      return res.status(400).send(`Error: ${err.message}`);
+    } else if (err.name && err.name === 'TokenExpiredError') {
+      return res.status(400).send('Error: The time window to connect your GitHub account has expired. Please return to Slack to re-start the process of connecting your GitHub account.');
+    }
+    throw err;
+  }
+
+  const {
+    trigger_id,
+    teamSlackId,
+    teamSlackDomain,
+    userSlackId,
+    channelSlackId,
+    replaySlashCommand,
+    actionCallbackId,
+  } = state;
+
+  // Set session cookie with oauth state that we can verify it in the OAuth callback
+  const githubOAuthState = crypto.randomBytes(Math.ceil(30 / 2)).toString('hex').slice(0, 30);
```

Comment:
```suggestion
  const githubOAuthState = crypto.randomBytes(15).toString('hex');
```

Since `Math.ceil(30 / 2)` is constant, why not putting it directly? 😄  Also I've checked that the result is always a 30 chars string, so there's no need to do the `slice()`. Also, interesting use of `slice()`, I would have used `substring()` since it's a string 🤔

Reply:
Not being the author, I guess the intend was documentation, because one can see how the 30 and the 30 match. So it reads like: "we want 30 chars, first we take 15 random bytes, then we hex encode them to make them 30 (since each byte is 00..FF) and in order to guarantee that we always create a valid token, we slice 30, no matter what" ... It's a nice story that the code is telling. However, it doesn't really make sense, so 
`crypto.randomBytes(15).toString('hex');` is way better 😄

Reply:
> Also, interesting use of `slice()`, I would have used `substring()` since it's a string 🤔
Fun fact, I think `slice` has the more appropriate behaviour. `substring` is marginally faster than slice (depending on browser details). 
Here is a comparison: https://stackoverflow.com/a/2243835/2375386. I think the argument swapping of substring is horrible. But its just opinion, so never mind. 🤷‍♂️

Reply:
All great points 💯 

For context, this originally came from here: https://github.com/integrations/slack/pull/264#discussion_r167062267